## SQL QUERIES

In [18]:
import pandas as pd
import sqlite3

In [19]:
df = pd.read_excel("data/telecom_churn.xlsx")
conn = sqlite3.connect("churn.db")
df.to_sql("customers", conn, if_exists="replace", index=False)

3333

In [20]:
query="""
SELECT * FROM customers
LIMIT 5
"""
result = pd.read_sql(query, conn)
print(result)

   Churn  AccountWeeks  ContractRenewal  DataPlan  DataUsage  CustServCalls  \
0      0           128                1         1        2.7              1   
1      0           107                1         1        3.7              1   
2      0           137                1         0        0.0              0   
3      0            84                0         0        0.0              2   
4      0            75                0         0        0.0              3   

   DayMins  DayCalls  MonthlyCharge  OverageFee  RoamMins  
0    265.1       110           89.0        9.87      10.0  
1    161.6       123           82.0        9.78      13.7  
2    243.4       114           52.0        6.06      12.2  
3    299.4        71           57.0        3.10       6.6  
4    166.7       113           41.0        7.42      10.1  


In [21]:
query_1="""
SELECT count(*) AS TOTAL_CUSTOMERS FROM customers
"""
result=pd.read_sql(query_1,conn)
print(result)

   TOTAL_CUSTOMERS
0             3333


In [22]:
query_2= """
SELECT COUNT(*) AS churned_customers
FROM customers
WHERE Churn = 1;
"""
result = pd.read_sql(query_2, conn)
print(result)

   churned_customers
0                483


In [23]:
query_3= """
SELECT
(COUNT(CASE WHEN Churn = 1 THEN 1 END) * 100.0 / COUNT(*))
AS churn_rate
FROM customers;
"""
result = pd.read_sql(query_3, conn)
print(result)

   churn_rate
0   14.491449


In [24]:
query_4= """
SELECT
Churn,
AVG(MonthlyCharge) AS avg_monthly_charge
FROM customers
GROUP BY Churn;
"""
result = pd.read_sql(query_4, conn)
print(result)

   Churn  avg_monthly_charge
0      0           55.816246
1      1           59.190062


In [25]:
query_5= """
SELECT
Churn,
AVG(CustServCalls) AS avg_service_calls
FROM customers
GROUP BY Churn;
"""
result = pd.read_sql(query_5, conn)
print(result)

   Churn  avg_service_calls
0      0           1.449825
1      1           2.229814


In [26]:
query_6= """
SELECT
ContractRenewal,
COUNT(*) AS total_customers,
SUM(CASE WHEN Churn = 1 THEN 1 ELSE 0 END) AS churned_customers
FROM customers
GROUP BY ContractRenewal;
"""
result = pd.read_sql(query_6, conn)
print(result)

   ContractRenewal  total_customers  churned_customers
0                0              323                137
1                1             3010                346


In [27]:
query_7= """
SELECT
Churn,
AVG(DataUsage) AS avg_data_usage
FROM customers
GROUP BY Churn;
"""
result = pd.read_sql(query_7, conn)
print(result)

   Churn  avg_data_usage
0      0        0.862151
1      1        0.546957


In [28]:
query_8= """
SELECT *
FROM customers
WHERE CustServCalls > 3
AND MonthlyCharge > 70;
"""
result = pd.read_sql(query_8, conn)
print(result)

    Churn  AccountWeeks  ContractRenewal  DataPlan  DataUsage  CustServCalls  \
0       1           161                1         0       0.00              4   
1       0            94                0         1       2.57              4   
2       1           131                0         1       3.73              4   
3       0           104                1         1       3.59              4   
4       1            33                1         1       3.75              4   
5       0           106                1         1       3.81              4   
6       0           114                1         1       2.27              4   
7       0           179                0         1       2.73              4   
8       0            79                1         1       1.70              4   
9       1           118                1         1       2.59              5   
10      0           122                1         1       2.48              7   
11      1           152                0

In [29]:
query_9 = """
SELECT
Churn,
AVG(RoamMins) AS avg_roaming_minutes
FROM customers
GROUP BY Churn;
          """
result = pd.read_sql(query_9, conn)
print(result)

   Churn  avg_roaming_minutes
0      0            10.158877
1      1            10.700000


## OVERALL INSIGHTS FROM QUERIES

1.Total Customer Base

The dataset contains 3,333 customers. This represents the total number of telecom users analyzed in the study.

Business meaning:
This dataset provides a moderate-sized sample to analyze customer behavior and identify churn patterns.

2.Customer Churn Rate

A portion of the customers have churned (left the telecom service). The churn rate represents the percentage of customers who discontinued the service.

Business meaning:
Understanding the churn rate helps telecom companies estimate revenue loss and identify retention opportunities.

3.Monthly Charges and Churn

Customers who churn tend to have higher average monthly charges compared to customers who stay.

Business insight:
Higher pricing may increase the likelihood of churn. Competitive pricing or targeted discounts may help reduce customer loss.

4.Customer Service Complaints

Customers who churn have more customer service calls on average.

Business insight:
Frequent complaints indicate dissatisfaction with the service. Improving customer support and resolving issues faster could reduce churn.

5.Contract Renewal Impact

Customers who did not renew their contracts show significantly higher churn compared to those who renewed.

Business insight:
Encouraging contract renewals through loyalty programs or incentives could improve retention.

6.Data Usage Patterns

Customers who churn show different data usage patterns compared to retained customers.

Business insight:
Understanding how customers use data can help telecom companies design better plans tailored to customer needs.

7.High-Risk Customers

Customers with More than 3 customer service calls and Monthly charges above 70 appear to be high-risk churn candidates.

Business insight:
These customers should be targeted with proactive retention strategies such as special offers or improved support.

8.Roaming Usage and Churn

Customers with higher roaming minutes show a slightly higher tendency to churn.

Business insight:
Roaming charges or service quality during travel might influence customer satisfaction.